pennylane_lightning/src/tests/CompareVector.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cstdlib>
+#include <vector>
+
+template <class T, class AllocA, class AllocB>
+bool operator==(const std::vector<T, AllocA> &lhs,
+                const std::vector<T, AllocB> &rhs) {
+    if (lhs.size() != rhs.size()) {
+        return false;
+    }
+    for (size_t idx = 0; idx < lhs.size(); idx++) {
+        if (lhs[idx] != rhs[idx]) {
+            return false;
+        }
+    }
+    return true;
+}
```

Comment:
This is only used inside the test code. To test approximation in test code, `PLApprox` already exists.

Reply:
@AmintorDusko This is used somewhere inside 'Test_GateImplementations_Nonparam.cpp' file. Adding this definition inside of `TestHelper.hpp` didn't work due to some complex relationship between macros and operator overloading.

Reply:
Can we have this overload template scope limited to integers, restricted to the `Test_GateImplementations_Nonparam.cpp` file, or both?

Reply:
May I ask why? We already use this function for float and double types and different allocators. I also think I may use it in other files when needed.

Reply:
This is just my opinion, please disregard it if this doesn't make sense.
1. If we have a header called `CompareVector.hpp` I expect all comparisons between vectors to be there.
2. If someone inadvertently includes this header and tries to compare two floating-point vectors this will simplify return false. This can a headache to debug, especially for someone not so close to all this development.

Reply:
I note that there is already `operator==` defined for `std::vector` in the standard library (see https://en.cppreference.com/w/cpp/container/vector/operator_cmp and https://godbolt.org/z/6Kbjhvjas). This is just an extension of it for different allocators.

Reply:
Yes. You are right about that. Thanks for the clarification.